### PR TITLE
#110 Scan for nearest beacon

### PR DESCRIPTION
**Pull request by JasonLighthunter**

### CODE REVIEW

File: Bartimeus/ViewController.swift [hunk 1]
```diff
@@ -17,38 +17,29 @@ class ViewController: UIViewController {
...
-                    self.scanner = try MovinSDK.getBeaconScanner()
-                    let maps = try? MovinSDK.getMaps()
-                    self.scanner.add((maps?[0])!)
-                    self.scanner.start(with: BeaconScannerListener.instance)
-                } catch {
-                    print("No Bluetooth Enabled")
-                }
+        let path: String = Bundle.main.path(forResource: "config", ofType: "plist")!
+        let config: NSDictionary = NSDictionary(contentsOfFile: path)!
+        self.movinAPIKey = (config.object(forKey: "api-key") as? String)!
```
Comment (beschoenen) [inline Bartimeus/ViewController.swift]: Move this to the constructor and remove the default value of `self.movinAPIKey`

File: Bartimeus/ViewController.swift [hunk 2]
```diff
@@ -9,16 +9,40 @@
...
+            }
+        }
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
-
+
```
Comment (beschoenen) [inline Bartimeus/ViewController.swift]: Remove trailing whitespace

File: Bartimeus/ViewController.swift [hunk 3]
```diff
@@ -17,38 +17,29 @@ class ViewController: UIViewController {
...
+        MovinLocationManager.instance().requestWhenInUseAuthorization()
+
+        MovinSDK.setCacheProtocolForTypes(MovinCacheableData.all, to: MovinCacheProtocol.doNotCache())
+
+        MovinSDK.initWithCustomer(clientName, andApikey: self.movinAPIKey) { _ -> Void in
+            do {
+                self.scanner = try MovinSDK.getBeaconScanner()
+                let maps = try? MovinSDK.getMaps()
+                self.scanner.add((maps?[0])!)
```
Comment (beschoenen) [inline Bartimeus/ViewController.swift]: Can't this be simplified? `?` and `!`

Comment (JasonLighthunter) [inline Bartimeus/ViewController.swift]: bedoel je 
```swift
self.scanner.add(MovinSDK.getMaps())
```
of anders?

File: Bartimeus/ViewController.swift (hunk 1 above)
Comment (beschoenen) [inline Bartimeus/ViewController.swift]: Move this to the constructor and remove the default value of self.movinAPIKey

File: Bartimeus/ViewController.swift [hunk 4]
```diff
@@ -9,16 +9,36 @@
 import UIKit
 
 class ViewController: UIViewController {
+    let clientName = "bartimeus-c"
+    var movinAPIKey : String = ""
```
Comment (beschoenen) [inline Bartimeus/ViewController.swift]: I'd prefer just this: `var movinAPIKey : String` to minimize bugs.

File: Bartimeus/ViewController.swift [hunk 5]
```diff
@@ -9,16 +9,36 @@
...
+
+    private var scanner: MovinBeaconScanner!
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+
+        let path: String = Bundle.main.path(forResource: "config", ofType: "plist")!
+        let config: NSDictionary = NSDictionary(contentsOfFile: path)!
+        self.movinAPIKey = (config.object(forKey: "api-key") as? String)!
```
Comment (beschoenen) [inline Bartimeus/ViewController.swift]: These 3 lines should be moved to an init method

File: Bartimeus/ViewController.swift (hunk 4 above)
Comment (JasonLighthunter) [inline Bartimeus/ViewController.swift]: Xcode will dit zo

**Pull request merged**